gridstatus/isone_api/isone_api.py

```diff
@@ -382,3 +339,200 @@ def get_dayahead_hourly_demand(
         df = df.drop_duplicates(subset=["BeginDate", "Location"], keep="first")
 
         return self._handle_demand(df, interval_minutes=60)
+
+    def _handle_demand(
+        self,
+        df: pd.DataFrame,
+        interval_minutes: int = 60,
+    ) -> pd.DataFrame:
+        """
+        Process demand DataFrame: convert types, rename columns, and add Interval End.
+
+        Args:
+            df (pd.DataFrame): Input DataFrame with demand data.
+            interval_minutes (int): Duration of each interval in minutes. Default is 60.
+
+        Returns:
+            pd.DataFrame: Processed DataFrame.
+        """
+        try:
+            # Try the standard pandas datetime conversion first
+            df["Interval Start"] = pd.to_datetime(df["BeginDate"])
+            df["Interval Start"] = df["Interval Start"].dt.tz_convert(
+                self.default_timezone,
+            )
+        except AttributeError:
+            # NOTE(kladar) consistently the above logic fails for DST conversion days.
+            # This catches those and fixes it.
+            # Data coming out looks good, no missed intervals and no duplicates.
+            log.warning("Standard datetime conversion failed. Using custom parsing.")
+            df["Interval Start"] = df["BeginDate"].apply(
+                self.parse_problematic_datetime,
+            )
+
+        df = df.sort_values(["Interval Start", "Location"])
+        df["Interval End"] = df["Interval Start"] + pd.Timedelta(
+            minutes=interval_minutes,
+        )
+
+        df["Load"] = pd.to_numeric(df["Load"], errors="coerce")
+        df["Location Id"] = pd.to_numeric(df["LocId"], errors="coerce")
+
+        log.info(
+            f"Processed demand data: {len(df)} entries from {df['Interval Start'].min()} to {df['Interval Start'].max()}",
+        )
+        return df[["Interval Start", "Interval End", "Location", "Location Id", "Load"]]
+
+    @support_date_range("DAY_START")
+    def get_hourly_load_forecast(
+        self,
+        date: str | pd.Timestamp = "latest",
+        end: str | pd.Timestamp | None = None,
+        horizons: Literal["latest", "all"] = "all",
```

Comment:
Given the explanation for `horizons`, I wonder if it makes sense to set the default to `"latest"`. My intuition is most people would only want the most recent forecast.

Reply:
I actually had it that way and thought I wouldn't hide the fact that you can get all of them, given that `latest` is a subset of `all`. I see `all` as the critical functionality and `latest` is nice-to-have.